src/main.rs

```diff
@@ -209,5 +208,57 @@ async fn event_handler(
 			.await;
 	}
 
+	if let serenity::FullEvent::Ready { .. } = event {
+		let http = ctx.http.clone();
+		tokio::spawn(init_server_icon_changer(http, data.discord_guild_id));
+	}
+
 	Ok(())
 }
+
+async fn init_server_icon_changer(ctx: impl serenity::CacheHttp, guild_id: serenity::GuildId) {
+	let mut interval = tokio::time::interval(Duration::from_secs(
+		rand::thread_rng().gen_range((24 * 60 * 60)..(48 * 60 * 60)),
```

Comment:
This won't choose a random time for each interval. I'd instead just use sleep inside the loop